airbyte-integrations/connectors/destination-snowflake/src/test/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestinationTest.java

```diff
@@ -53,4 +76,54 @@ public void useInsertStrategyTest() {
     assertFalse(SnowflakeDestination.isS3Copy(stubConfig));
   }
 
+  @Test
+  public void testCleanupStageOnFailure() throws Exception {
+
+    JdbcDatabase mockDb = mock(JdbcDatabase.class);
+    SnowflakeStagingSqlOperations sqlOperations = mock(SnowflakeStagingSqlOperations.class);
+    final var testMessages = generateTestMessages();
+    final JsonNode config = Jsons.deserialize(IOs.readFile(Path.of("secrets/insert_config.json")));
+
+    AirbyteMessageConsumer airbyteMessageConsumer = new SnowflakeInternalStagingConsumerFactory()
+            .create(Destination::defaultOutputRecordCollector, mockDb,
+            sqlOperations, new SnowflakeSQLNameTransformer(), config, getCatalog());
+    doThrow(SQLException.class).when(sqlOperations).copyIntoTmpTableFromStage(any(),anyString(),anyString(),anyString());
+
+    airbyteMessageConsumer.start();
+    for (AirbyteMessage m : testMessages) {
+        airbyteMessageConsumer.accept(m);
+    }
+
+    try {
+      airbyteMessageConsumer.close();
+    }catch (Exception e){
```

Comment:
does this need to catch all exception, or can it only catch `SQLException`? (in retrospect, same question here https://github.com/airbytehq/airbyte/pull/9141/files#diff-33bfb48d5bb53941a89b564d1e0819ad2a78ef0c19dcb169d568e2205a19959aR181 )

Reply:
> does this need to catch all exception, or can it only catch `SQLException`? (in retrospect, same question here https://github.com/airbytehq/airbyte/pull/9141/files#diff-33bfb48d5bb53941a89b564d1e0819ad2a78ef0c19dcb169d568e2205a19959aR181 )

really good question. I believe SQLException should be enough